bitcoin/src/blockdata/script.rs

```diff
@@ -1222,6 +1236,66 @@ mod test {
     }
 
+    #[test]
+    fn test_p2pk_pubkey_bytes() {
+        // happy path
+        let key_str = "0411db93e1dcdb8a016b49840f8c53bc1eb68a382e97b1482ecad7b148a6909a5cb2e0eaddfb84ccf9744464f82e160bfa9b8b64f9d4c03f999b8643f656b412a3";
+        let key = PublicKey::from_str(key_str).unwrap();
+        let p2pk = Script::builder().push_key(&key).push_opcode(OP_CHECKSIG).into_script();
+        let actual = p2pk.p2pk_pubkey_bytes().unwrap();
+        assert_eq!(actual.to_vec(), key.to_bytes());
+
+        // invalid script checks
+        let no_checksig = Script::builder().push_key(&key).into_script();
+        assert_eq!(no_checksig.p2pk_pubkey_bytes(), None);
+        let empty_script = Script::builder().into_script();
+        assert!(empty_script.p2pk_pubkey_bytes().is_none());
+        let no_push_bytes = Script::builder().push_opcode(OP_CHECKSIG).into_script();
+        assert!(no_push_bytes.p2pk_pubkey_bytes().is_none());
+
+        // invalid key check
+        let malformed_key = "21032e58afe51f9ed8ad3cc7897f634d881fdbe49816429ded8156bebd2ffd1";
+        let invalid_p2pk_script = Script::builder().push_slice(malformed_key.as_bytes()).into_script();
+        assert!(invalid_p2pk_script.p2pk_pubkey_bytes().is_none());
+
+        // this key is the compressed form of the first key in this test
+        let compressed_key_str = "0311db93e1dcdb8a016b49840f8c53bc1eb68a382e97b1482ecad7b148a6909a5c";
+        let key = PublicKey::from_str(compressed_key_str).unwrap();
+        let p2pk = Script::builder().push_key(&key).push_opcode(OP_CHECKSIG).into_script();
+        let actual = p2pk.p2pk_pubkey_bytes().unwrap();
+        assert_eq!(actual.to_vec(), key.to_bytes());
+    }
+
+    #[test]
+    fn test_p2pk_public_key() {
+        // happy path
+        let key_str = "0411db93e1dcdb8a016b49840f8c53bc1eb68a382e97b1482ecad7b148a6909a5cb2e0eaddfb84ccf9744464f82e160bfa9b8b64f9d4c03f999b8643f656b412a3";
+        let key = PublicKey::from_str(key_str).unwrap();
+        let p2pk = Script::builder().push_key(&key).push_opcode(OP_CHECKSIG).into_script();
+        let actual = p2pk.p2pk_public_key().unwrap();
+        assert_eq!(actual, key);
+
+        // invalid script checks
+        let no_checksig = Script::builder().push_key(&key).into_script();
+        assert_eq!(no_checksig.p2pk_public_key(), None);
+        let empty_script = Script::builder().into_script();
+        assert!(empty_script.p2pk_public_key().is_none());
+        let no_push_bytes = Script::builder().push_opcode(OP_CHECKSIG).into_script();
+        assert!(no_push_bytes.p2pk_public_key().is_none());
+
+        // check invalid key
+        let malformed_key = "21032e58afe51f9ed8ad3cc7897f634d881fdbe49816429ded8156bebd2ffd1";
+        let invalid_p2pk_script = Script::builder().push_slice(malformed_key.as_bytes()).into_script();
+        assert!(invalid_p2pk_script.p2pk_public_key().is_none());
```

Comment:
This test is missing `OP_CHECKSIG` which makes it bogus.

Reply:
It's testing if a script is malformed, should this case be excluded because its tested/covered in the test for `p2pk_pubkey_bytes`?

Reply:
No, it's testing different things. When the script is valid in all aspects *except for the key*:

* `p2pk_pubkey_bytes()` should return `Some(_)`
* `p2pk_public_key()` should return `None`

These are not actually tested because the script is invalid due to missing `OP_CHECKSIG`.

Reply:
Ok, I'm going through these tests and making sure that all cases are covered.